Makefile:14: recipe for target 'init' failed

##### Steps to reproduce the problem:

1. make init
2.
3.


##### Any other comments? What have you tried so far?

	$ ~/chomper$ make init
	/bin/bash: pipenv: command not found
	pip install pipenv # Install pipenv
	Collecting pipenv
	Collecting pathlib (from pipenv)
	Collecting setuptools>=36.2.1virtualenv (from pipenv)
	  Using cached setuptools-38.5.2-py2.py3-none-any.whl
	Collecting virtualenv-clone>=0.2.5 (from pipenv)
	  Using cached virtualenv_clone-0.3.0-py2.py3-none-any.whl
	Installing collected packages: pathlib, setuptools, virtualenv-clone, pipenv
	Successfully installed pathlib-1.0.1 pipenv-11.1.6 setuptools-38.5.2 virtualenv-clone-0.3.0
	pipenv install --dev # Install packages
	/bin/bash: pipenv: command not found
	Makefile:14: recipe for target 'init' failed
	make: *** [init] Error 127

Not sure how to resolve this "Install packages" error.


##### System information

	$ lsb_release -a

	Distributor ID:	Debian
	Description:	Debian GNU/Linux 9.3 (stretch)
	Release:	9.3
	Codename:	stretch

Hey, I'm also having this problem using the same version of debian. I changed the `pipenv` line in the Makefile as you suggested. Any suggestions?

	$ make init

	sudo -H pip install -U pipenv # Install pipenv
	Requirement already up-to-date: pipenv in /usr/local/lib/python2.7/dist-packages
	Requirement already up-to-date: virtualenv-clone>=0.2.5 in /usr/local/lib/python2.7/dist-packages (from pipenv)
	Requirement already up-to-date: setuptools>=36.2.1 in /usr/local/lib/python2.7/dist-packages (from pipenv)
	Requirement already up-to-date: pathlib in /usr/local/lib/python2.7/dist-packages (from pipenv)
	Requirement already up-to-date: virtualenv in /usr/local/lib/python2.7/dist-packages (from pipenv)
	pipenv install --dev # Install packages
	Warning: Python 3.6 was not found on your system…
	You can specify specific versions of Python with:
	  $ pipenv --python path/to/python
	/home/user/.local/lib/python2.7/site-packages/pipenv/utils.py:1152: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/pipenv-zflZBs-requirements'>
	  warnings.warn(warn_message, ResourceWarning)
	Makefile:14: recipe for target 'init' failed
	make: *** [init] Error 1


	$ whereis python3

	python3: /usr/bin/python3 /usr/bin/python3.5m-config /usr/bin/python3.5m /usr/bin/python3.5-config /usr/bin/python3.5 /usr/lib/python3 /usr/lib/python3.5 /etc/python3 /etc/python3.5 /usr/local/lib/python3.5 /usr/include/python3.5m /usr/include/python3.5 /usr/share/python3 /usr/share/man/man1/python3.1.gz


	$ cat Makefile 

	# Copyright: (c) 2018, Aniket Panjwani <aniket@addictedto.tech>
	# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)

	SHELL:=/bin/bash
	INTERPRETER=$(shell pipenv --py)
	CURRENT_USER:=$(shell whoami)
	CURRENT_DIR:=$(shell pwd)
	SUDOERS_FILE:=/etc/sudoers.d/chomper
	SUDOERS_ENTRY_MIDDLE:=" ALL=(ALL) NOPASSWD: "
	SUDOERS_ENTRY_SUFFIX:="/chomper/block.py *"
	SUDOERS_ENTRY:=$(CURRENT_USER)$(SUDOERS_ENTRY_MIDDLE)$(INTERPRETER) $(CURRENT_DIR)$(SUDOERS_ENTRY_SUFFIX)

	init:
		sudo -H pip install -U pipenv # Install pipenv
		pipenv install --dev # Install packages
		pipenv run mitmdump & sleep 2 && kill -9 $$! # Generate certificates
		openssl x509 -outform der -in ~/.mitmproxy/mitmproxy-ca.pem -out ~/.mitmproxy/mitmproxy-ca.crt
		sudo cp /home/$(CURRENT_USER)/.mitmproxy/mitmproxy-ca.crt /usr/local/share/ca-certificates/mitmproxy-ca.crt # Install root certificates
		sudo update-ca-certificates
		sudo sh ./chomper/certs.sh # Make browsers recognize root certificates
		sudo sysctl -w net.ipv4.ip_forward=1 # Enable ipv4 forwarding
		sudo sysctl -w net.ipv6.conf.all.forwarding=1 # Enable ipv6 forwarding
		sudo sysctl -p # Lock in new ip forwarding settings.

	lock:
		echo $(SUDOERS_ENTRY) | sudo EDITOR='tee' visudo -f $(SUDOERS_FILE)
		sudo chown -R root:root ./
		sudo chown $(CURRENT_USER):$(CURRENT_USER) ./data/rules.yaml
		sudo chown $(CURRENT_USER):$(CURRENT_USER) ./readme.org
		sudo chown $(CURRENT_USER):$(CURRENT_USER) ./.gitignore
		sudo chown -R $(CURRENT_USER):$(CURRENT_USER) ./.git

	reset:
		sudo env PATH=$(PATH):/usr/sbin:/sbin ${INTERPRETER}3.6m ${CURRENT_DIR}/chomper/reset.py

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ init:
 
   # Install certificates
 	pipenv run mitmdump & sleep 2 && kill -9 $$! # Generate certificates
-	openssl x509 -outform der -in ~/.mitmproxy/mitmproxy-ca.pem -out ~/.mitmproxy/mitmproxy-ca.crt
+	openssl x509 -outform der -in /home/$(CURRENT_USER)/.mitmproxy/mitmproxy-ca.pem -out /home/$(CURRENT_USER)/.mitmproxy/mitmproxy-ca.crt
 	sudo cp /home/$(CURRENT_USER)/.mitmproxy/mitmproxy-ca.crt /usr/local/share/ca-certificates/mitmproxy-ca.crt # Install root certificates
 	sudo update-ca-certificates
 	sudo sh ./chomper/certs.sh # Make browsers recognize root certificates

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 # Copyright: (c) 2018, Aniket Panjwani <aniket@addictedto.tech>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
+.PHONY: init lock reset
 SHELL:=/bin/bash
 INTERPRETER=$(shell pipenv --py)
 CURRENT_USER:=$(shell whoami)

--- a/Makefile
+++ b/Makefile
@@ -15,8 +15,9 @@ init:
   # Install pyenv and put it on PATH
 	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
 	echo "export PATH='$$HOME/.pyenv/bin:$$PATH'" >> /home/$(CURRENT_USER)/.bashrc
-	echo "eval '$$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
-	source ~/.bashrc
+	source /home/$(CURRENT_USER)/.bashrc
+	echo "eval '$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
+	source /home/$(CURRENT_USER)/.bashrc
 
   # Install Python 3.6.4
 	pyenv update

--- a/Makefile
+++ b/Makefile
@@ -40,9 +40,6 @@ init:
 	sudo sysctl -w net.ipv6.conf.all.forwarding=1 # Enable ipv6 forwarding
 	sudo sysctl -p # Lock in new ip forwarding settings.
 
-  # Put chomper executable on PATH
-	echo "export PATH='$$HOME/.pyenv/bin:$$PATH'" >> /home/$(CURRENT_USER)/.bashrc
-
 lock:
 	echo $(SUDOERS_ENTRY) | sudo EDITOR='tee' visudo -f $(SUDOERS_FILE)
 	sudo chown -R root:root ./

--- a/Makefile
+++ b/Makefile
@@ -40,6 +40,9 @@ init:
 	sudo sysctl -w net.ipv6.conf.all.forwarding=1 # Enable ipv6 forwarding
 	sudo sysctl -p # Lock in new ip forwarding settings.
 
+  # Put chomper executable on PATH
+	echo "export PATH='$$HOME/.pyenv/bin:$$PATH'" >> /home/$(CURRENT_USER)/.bashrc
+
 lock:
 	echo $(SUDOERS_ENTRY) | sudo EDITOR='tee' visudo -f $(SUDOERS_FILE)
 	sudo chown -R root:root ./

--- a/Makefile
+++ b/Makefile
@@ -11,13 +11,33 @@ SUDOERS_ENTRY_SUFFIX:="/chomper/block.py *"
 SUDOERS_ENTRY:=$(CURRENT_USER)$(SUDOERS_ENTRY_MIDDLE)$(INTERPRETER) $(CURRENT_DIR)$(SUDOERS_ENTRY_SUFFIX)
 
 init:
-	pip install pipenv # Install pipenv
-	pipenv install --dev # Install packages
+  # Get build tools
+	sudo apt-get update
+	sudo apt-get install curl git build-essential zlib1g-dev libbz2-dev libsqlite3-dev libreadline-dev libncurses5-dev libssl-dev libgdbm-dev
+
+  # Install pyenv and put it on PATH
+	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+	echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
+	echo 'eval "$(pyenv init -)"' >> ~/.bashrc
+	source ~/.bashrc
+
+  # Install Python 3.6.4
+	pyenv update
+	pyenv install 3.6.4
+	pyenv local 3.6.4
+
+  # Install pipenv and virtual environment
+	sudo -H pip install -U pipenv # Install pipenv
+	pipenv install --dev --python 3.6.4 # Install packages
+
+  # Install certificates
 	pipenv run mitmdump & sleep 2 && kill -9 $$! # Generate certificates
 	openssl x509 -outform der -in ~/.mitmproxy/mitmproxy-ca.pem -out ~/.mitmproxy/mitmproxy-ca.crt
 	sudo cp /home/$(CURRENT_USER)/.mitmproxy/mitmproxy-ca.crt /usr/local/share/ca-certificates/mitmproxy-ca.crt # Install root certificates
 	sudo update-ca-certificates
 	sudo sh ./chomper/certs.sh # Make browsers recognize root certificates
+
+  # Enable ip forwarding
 	sudo sysctl -w net.ipv4.ip_forward=1 # Enable ipv4 forwarding
 	sudo sysctl -w net.ipv6.conf.all.forwarding=1 # Enable ipv6 forwarding
 	sudo sysctl -p # Lock in new ip forwarding settings.

--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ SUDOERS_ENTRY:=$(CURRENT_USER)$(SUDOERS_ENTRY_MIDDLE)$(INTERPRETER) $(CURRENT_DI
 
 init:
   # Install pyenv and put it on PATH
-  curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
 	echo "export PATH='$$HOME/.pyenv/bin:$$PATH'" >> /home/$(CURRENT_USER)/.bashrc
 	echo "eval '$$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
 	source ~/.bashrc

--- a/Makefile
+++ b/Makefile
@@ -11,14 +11,10 @@ SUDOERS_ENTRY_SUFFIX:="/chomper/block.py *"
 SUDOERS_ENTRY:=$(CURRENT_USER)$(SUDOERS_ENTRY_MIDDLE)$(INTERPRETER) $(CURRENT_DIR)$(SUDOERS_ENTRY_SUFFIX)
 
 init:
-  # Get build tools
-	sudo apt-get update
-	sudo apt-get install curl git build-essential zlib1g-dev libbz2-dev libsqlite3-dev libreadline-dev libncurses5-dev libssl-dev libgdbm-dev
-
   # Install pyenv and put it on PATH
 	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-	echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
-	echo 'eval "$(pyenv init -)"' >> ~/.bashrc
+	echo "export PATH='\$HOME/.pyenv/bin:\$PATH'" >> /home/$(CURRENT_USER)/.bashrc
+	echo "eval '\$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
 	source ~/.bashrc
 
   # Install Python 3.6.4

--- a/Makefile
+++ b/Makefile
@@ -12,9 +12,9 @@ SUDOERS_ENTRY:=$(CURRENT_USER)$(SUDOERS_ENTRY_MIDDLE)$(INTERPRETER) $(CURRENT_DI
 
 init:
   # Install pyenv and put it on PATH
-	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-	echo "export PATH='\$HOME/.pyenv/bin:\$PATH'" >> /home/$(CURRENT_USER)/.bashrc
-	echo "eval '\$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
+  curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+	echo "export PATH='$$HOME/.pyenv/bin:$$PATH'" >> /home/$(CURRENT_USER)/.bashrc
+	echo "eval '$$(pyenv init -)'" >> /home/$(CURRENT_USER)/.bashrc
 	source ~/.bashrc
 
   # Install Python 3.6.4

